src/utilities/color-transformers.ts

```diff
@@ -314,3 +325,38 @@ export function colorToHsla(color: string): HSLAColor {
       );
   }
 }
+
+export function normalizeColorString(value: string) {
+  return value.toLowerCase().replace(/\s/g, '');
+}
+
+export const hexToHsb: (hex: string) => HSBColor = compose(
+  rgbToHsb,
+  hexToRgb,
+);
+
+export function expandHex(hex: string) {
+  if (hex.length === 4) {
+    return `#${hex[1]}${hex[1]}${hex[2]}${hex[2]}${hex[3]}${hex[3]}`;
+  }
+  return hex;
+}
+
+export function nameToHex(value: string) {
+  const hex = nameHexMap[normalizeColorString(value)];
+  return expandHex(`#${hex}`);
+}
+
+export function rgbStringToHex(value: string) {
+  const rgb = normalizeColorString(value).match(RGB_STRING_TO_HEX_REGEX) || [
+    'rgb(0,0,0)',
```

Comment:
```suggestion
   undefined,
```

Since we don't use the full capture